### PR TITLE
Add curl user-agent to retrieve favicon

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/1380

### CODE REVIEW

File: lib/favicons.php [hunk 1]
```diff
@@ -18,11 +18,12 @@ function download_favicon($website, $dest) {
 		return @copy($default_favicon, $dest);
 	}
 
-	syslog(LOG_DEBUG, 'FreshRSS Favicon GET ' . $favicon_url);
+	syslog(LOG_INFO, 'FreshRSS Favicon GET ' . $favicon_url);
 	$c = curl_init($favicon_url);
 	curl_setopt($c, CURLOPT_HEADER, false);
 	curl_setopt($c, CURLOPT_RETURNTRANSFER, true);
 	curl_setopt($c, CURLOPT_BINARYTRANSFER, true);
+	curl_setopt($c, CURLOPT_USERAGENT, 'FreshRSS/' . FRESHRSS_VERSION . ' (' . PHP_OS . '; ' . FRESHRSS_WEBSITE . ')');
```
Comment (kevinpapst) [inline lib/favicons.php:26]: Probably doesn't hurt, but I am against exposing the operating system of the webserver. 
Could that line be in the config file to allow users to overwrite the value?

File: lib/favicons.php [hunk 2]
```diff
@@ -9,7 +9,7 @@
 function download_favicon($website, $dest) {
 	global $favicons_dir, $default_favicon;
 
-	syslog(LOG_DEBUG, 'FreshRSS Favicon discovery GET ' . $website);
+	syslog(LOG_INFO, 'FreshRSS Favicon discovery GET ' . $website);
```
Comment (kevinpapst) [inline lib/favicons.php:12]: This one could stay debug, as there is already one info log for the same procedure below

Comment (Alkarex) [reply lib/favicons.php:12]: Yes, but not with the same address (two distinct requests) :-)

File: lib/favicons.php (hunk 1 above)
Comment (kevinpapst) [reply lib/favicons.php:26]: Well my comment was indeed without any background knowledge how FreshRSS handles this in all the other requests. Probably that should be streamlined (and configurable for the future).
Actually I just thought I give a little back to a great open source project by doing a code review ... 
I know that browsers spread the useragent information even further (that why I use extensions to switch them) but leaking server side information to other sites is probably not the best idea. Every admin/devop I know (me included) tries to hide every possible information about server-side OS and software versions (like type of webserver, php version and so on). 
Just my 2 cents - like in every code review - sharing thoughts is the most important part :)

File: lib/favicons.php (hunk 2 above)
Comment (kevinpapst) [reply lib/favicons.php:12]: I just imagined what happens in your log when you import an OPML with 200+ feeds ;-)
In most cases the favicon will be hosted on the same domain like the feed, so I would say this information is likely redundant - but you are right - these are two different urls. And this action won't happen to often^^

Comment (Alkarex) [reply lib/favicons.php:12]: We should indeed improve the logging options. I think it is important to have logs for the outgoing queries.

**Pull request merged**